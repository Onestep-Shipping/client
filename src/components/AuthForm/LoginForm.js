import './AuthForm.css';

/* eslint-disable no-unused-vars */
import React, {useCallback, useState} from 'react';

import Input from '../Input/Input';
import PropTypes from 'prop-types';
import app from '../../firebase/base.js';

const LoginForm = (props) => {
  const { closeModal } = props;

  const [errors, setErrors] = useState([]);
  const [displayErrors, setDisplayErrors] = useState(false);

  const handleLogin = useCallback(async (e) => {
    e.preventDefault();
    const email = e.target[0].value;
    const password = e.target[1].value;
    if (email.length > 0 && password.length > 0) {
      try {
        await app
          .auth()
          .signInWithEmailAndPassword(email, password);
        closeModal();
      } catch (error) {
        alert(error);
      }
    }
  }, [closeModal]);

  return (
    <div className="inner-container">
      <div className="header">
        <span className="schedule-header-text">Login</span>
      </div>
      <form className="box" onSubmit={handleLogin} noValidate>
        {errors.length > 0 &&
          errors.map((err, ind) => <small className="danger-error" key={ind}>{err}</small>)}
        <Input name="Email" type="email" displayErrors={displayErrors} />
        <Input name="Password" type="password" displayErrors={displayErrors} />
        <button className="login-btn">Login</button>
      </form>
    </div>
  );
};

export default LoginForm;

LoginForm.propTypes = {
  closeModal: PropTypes.func,
};
