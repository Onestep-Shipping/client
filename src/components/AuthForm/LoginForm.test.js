import LoginForm from './LoginForm';
import React from 'react';
import { shallow } from 'enzyme';

const closeModal = () => {}

let wrapped = shallow( <LoginForm closeModal={closeModal} />);

describe('LoginForm', () => {
  it('should render the LoginForm Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });
});