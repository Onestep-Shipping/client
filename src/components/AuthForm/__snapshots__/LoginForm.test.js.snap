// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render the LoginForm Component correctly 1`] = `
<div
  className="inner-container"
>
  <div
    className="header"
  >
    <span
      className="schedule-header-text"
    >
      Login
    </span>
  </div>
  <form
    className="box"
    noValidate={true}
    onSubmit={[Function]}
  >
    <Input
      displayErrors={false}
      name="Email"
      type="email"
    />
    <Input
      displayErrors={false}
      name="Password"
      type="password"
    />
    <button
      className="login-btn"
    >
      Login
    </button>
  </form>
</div>
`;
